### PR TITLE
Add sondehub amateur support

### PR DESCRIPTION
**Pull request by MedadRufus**
Work in progress still.
Plans:
1. add in option in config yml file to select option of uploading to sondehub

### CODE REVIEW

File: ttnhabbridge/src/main/java/nl/sikken/bertrik/hab/habitat/IHabitatRestApi.java [hunk 1]
```diff
@@ -1,5 +1,6 @@
 package nl.sikken.bertrik.hab.habitat;
 
+import nl.sikken.bertrik.hab.UploadResult;
```
Comment (MedadRufus) [inline ttnhabbridge/src/main/java/nl/sikken/bertrik/hab/habitat/IHabitatRestApi.java:3]: I don't think this import is needed

**Comment by LukePrior** [comment]
I'm not sure if I am reading it all correctly but it seems like you don't support some of the optional fields. 

- I think it would be a good idea to have uploader callsign as a configurable option with the default being `CHANGEME_TTNHABBRIDGE`. 
- I think the TTN/Helium API returns a list of the gateways that received the transmission so I think it could be handy to include the gateway with strongest signal uploading gateways position, rssi, and snr values in the `uploader_position`, `rssi`, and `snr` fields.
- You could also add the optional `modulation` field and set it as `LoRaWAN`.

I think it would be good to have the callsign as a setting under a new section of the `ttnhabbridge.yaml` file.

I also noticed that the `time_received` and `datetime` fields are using the same value, I'm not sure if the incoming TTN/Helium data includes the exact `time_received` as this might differ from the current time.

I guess you could also include the frequency of the payload as this also appears to be included in the TTN/Helium incoming data. 

You could also include what network the packet was received from TTN or Helium in the uploader_radio field.

**Comment by MedadRufus** [comment]
Sorry I meant to keep it a draft pull request while I worked on sondehub support. 
Yes things remaining left todo:

- [ ] correctly format `time_received` and `datetime` fields
- [ ] Add modulation option as Lorawan
- [ ] Add uploader information
- [ ] add frequency information

As for the uploader callsign setting, it must be handled as part of a separate issue/pull request. I have been wanting this feature as well.

**Comment by MedadRufus** [comment]
Still more work todo.

**Comment by MedadRufus** [comment]
opened issue #9 to deal with request for a configurable callsign.

**Comment by darksidelemm** [comment]
For the receiver information, I'd suggest uploading *all* receivers. Its always pretty cool to see all the receivers when a TTN payload is in the air.

Unfortunately the way sondehub works is that this means you will need to upload the telemetry frame multiple times, with the different receiver information (callsign, SNR, etc) in each frame. However, with gzip compression on the upload this doesnt really result in much extra data.

**Comment by MedadRufus** [comment]
got sidetracked from developing this feature - noted about zipping it all up.

**Comment by MedadRufus** [comment]
Note to self - consider splitting this pull request into:
1. add move common files for both habhub and sondehub to a common folder
2. add in sondehub support

This will make the pull request more focused on one thing at a time.

**Comment by MedadRufus** [comment]
make it such that the upload location(habitat or sondehub) is selected in the config file. No need to support both at the same time.

**Comment by bertrik** [comment]
I'm reading https://github.com/projecthorus/sondehub-amateur-tracker/wiki/Guide-for-HabHub-Users and see that apparently habhub is going away. So no need to support both habhub and sondehub I think.

**Comment by MedadRufus** [comment]
> I'm reading https://github.com/projecthorus/sondehub-amateur-tracker/wiki/Guide-for-HabHub-Users and see that apparently habhub is going away. So no need to support both habhub and sondehub I think.

True.

**Comment by darksidelemm** [comment]
Note that HabHub is getting switched off on the 31st of December.

**Pull request closed without merging**